pkg/avx/collector.go

```diff
@@ -24,6 +24,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"path/filepath"
+	"regexp"
```

Comment:
I just wish this could be done without regexp. We have been actively trying to keep out all regexp from file path/content parsers.

That said, I realize that currently we have no real alternative considering the different cgroup drivers and related container ID to path translation. However, let's make a mental note that once we have *proper explicit support* for both the systemd and the (lib)cgroupfs drivers, and consequently the necessary functions in place to convert back and forth between container IDs and cgroup paths, we'll revisit this and seriously consider eliminating regexp usage.

Reply:
> We have been actively trying to keep out all regexp from file path/content parsers.

OK, I didn't know this. How is the basename for the container path different between the cgroup drivers?

Reply:
There are several differences, IIRC. The systemd driver appends the .scope-suffix to directories, the other driver does not. The systemd driver also has an encoding/decoding scheme for certain characters in path names, the other driver does not have it. And I almost forgot: the container cgroup path also depends on the container runtime.

But since our current codebase lacks proper full support for the non-systemd cgroupfs driver and code to return a container ID for a path (exactly what you need here AFAICT), I think this regexp should not be a showstopper. I just wanted to make a note, so that we'll remember this in the future if we ever get the necessary functions to avoid relying on regexp here.